test/unit/style/style.test.js

```diff
@@ -2447,6 +2447,9 @@ test('Style#setProjection', (t) => {
         style.map._update = () => {};
         style.map.setProjection = (projection) => {
             style.map._explicitProjection = projection;
+            style.map._prioritizeAndUpdateProjection();
+        };
+        style.map._prioritizeAndUpdateProjection = () => {
```

Comment:
Nit: given the changes to the source, would it be possible to actually test  `_prioritizeAndUpdateProjection()` instead of mocking it as we do here? 

One idea would be to mock `map._updateProjection` to set an `explicitProjection` variable, and then check the set value after each call to `_prioritizeAndUpdateProjection`

As this logic is in `map`, maybe this would work better in `map.test.js`

Reply:
On second thought, this test is identical to this one in map.test.js: https://github.com/mapbox/mapbox-gl-js/blob/d821c49a4021859a8109c8f14020e2600a84b347/test/unit/ui/map.test.js#L1836-L1884

I'd lean toward removing this test entirely in favor of the more thorough version in map.test.js, not sure what I was thinking when I created the duplicate test!

Reply:
Oops! sounds good, they are identical! let's use the one in map.test.js since we don't have to mock the functions!